meshpy/mesh.py

```diff
@@ -76,6 +76,32 @@ def __init__(self):
         self.geometry_sets = GeometrySetContainer()
         self.boundary_conditions = BoundaryConditionContainer()
 
+    @staticmethod
+    def get_base_mesh_item_type(item):
+        """Return the base mesh type of the given item.
+
+        Amongst other things, we need this function so we can check if
+        all items of a list are of the same "base" type.
+        """
+        if isinstance(item, Mesh):
+            return Mesh
+        elif isinstance(item, Function):
+            return Function
+        elif isinstance(item, BoundaryConditionBase):
+            return BoundaryConditionBase
+        elif isinstance(item, Material):
+            return Material
+        elif isinstance(item, Node):
+            return Node
+        elif isinstance(item, Element):
+            return Element
+        elif isinstance(item, GeometrySetBase):
+            return GeometrySetBase
+        elif isinstance(item, GeometryName):
+            return GeometryName
+        else:
+            return type(item)
```

Comment:
Could we simplify this construct?

For example

```suggestion
        for cls in (Mesh, Function, BoundaryConditionBase, Material, Node, Element, GeometrySetBase, GeometryName):
            if isinstance(item, cls):
                return cls
        return type(item)
```